pulp_container/app/token_verification.py

```diff
@@ -61,7 +68,26 @@ def authenticate(self, request):
         if request.headers.get("Authorization") == "Basic Og==":
             return (AnonymousUser, None)
 
-        return super().authenticate(request)
+        try:
+            return super().authenticate(request)
+        except AuthenticationFailed:
+            if any(
+                self.PULP_AUTHENTICATION_CLASS in auth_class for auth_class in self.AUTH_CLASSES
+            ):
+                return RemoteUserRegistryAuthentication().authenticate(request)
+
+
+class RemoteUserRegistryAuthentication(RemoteUserAuthentication):
+    """
+    A class that authenticates users who were authenticated by a remote web server.
+    """
+
+    header = settings.REMOTE_USER_ENVIRON_NAME
+
+    def authenticate(self, request):
+        """Initialize an authenticated user based on the REMOTE_USER header."""
+        user = super().authenticate(request)
+        return (AnonymousUser, None) if user is None else user
```

Comment:
Is this needed? it seems like AnonymousUser is handled earlier in the code on line 68

Reply:
`AnonymousUser` is handled only for Basic authentication scenarios. We need to handle it here as well.

The solution comes from the following assumption. When a user could not be authenticated by a remote webserver or is trying to perform actions as an anonymous user, no `REMOTE_USER` header will be set by nginx.